WebChannelAPI.md

```diff
@@ -0,0 +1,474 @@
+
+        // Optionally store `meld` for direct access to the Meld Studio API.
+        const meld = channel.objects.meld;
+ 
+        // -------------------------
+        // Example Property Read:
+        // This dumps an object containing all Scenes, Layers, Effects, and Audio
+        // keyed by ID.
+        
+        console.log(meld.session.items);
+
+        // Status properties.
+        
+        console.log(meld.isStreaming);
+        console.log(meld.isRecording);
+
+        // -------------------------
+        // Example Listener:
+        // Subscribing to receive notifications when the Session property changes.
+        
+        meld.sessionChanged.connect(() => {
+          console.log("sessionChanged()");
+        });
+
+        // -------------------------
+        // Example Call:
+        // Will tell Meld Studio to toggle the recording state.
+        
+        meld.toggleRecord();
+      });
+    };
+
+   socket.onclose = function () {
+      console.log("Closed");
+   };
+
+   socket.onerror = function (error) {
+      console.log("Error: ", error);
+   };
+  }
+```
+
+## WebChannel Properties, Signals, and Slots
+
+### Properties
+
+| property    | type     | description                                                                                                                       |
+| ----------- | -------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| isStreaming | `bool`   | Indicates active streaming status.                                                                                                |
+| isRecording | `bool`   | Indicates active recording status.                                                                                                |
+| session     | `object` | Represents the current session state with a structured JSON object detailing scenes, tracks, layers, and effects. #session-schema |
+
+### Signals
+
+Signals provide methods for registering `listeners`.
+
+```js
+// Connect a signal callback.
+const ref = meld.signalName.connect((parameter: type) => {
+  // Do something with the information.
+});
+
+// Disconnect the signal callback.
+meld.signalName.disconnect(ref);
+```
+
+1. ## **`gainUpdated(trackId: string, gain: Number, muted: bool)`**
+
+   Called by Meld Studio to notify callbacks when the gain or mute status of a track changes.
+   These are significantly faster than waiting for the session to update and are ideal for real time volume controls.
+
+   **Parameters**:
+   |property|type|description
+   |---|---|---
+   |trackid|`string`|Identifier for the audio track.
+   |gain|`Number`|Current gain level.
+   |muted|`bool`|Mute status.
+
+   ```js
+   meld.gainUpdated.connect((trackId: string, gain: Number, muted: boolean) => {
+     // ... handle callback ...
+   });
+   ```
+
+2. ## **`sessionChanged`**
+
+   Triggered any time the contents of `session` change.
```

Comment:
We should introduce what the session represents somewhere.

Reply:
Fixed in 7d676dc136ad0ef8390ce78eeca8b749bd479623